### ISSUE TITLE
Privacy: ResizeObserver polyfill not downloaded

### ISSUE THREAD

**Issue by finnlueth** [squidfunk/mkdocs-material#5902]
### Bug description

Privacy Plugin misses built-in JS package.
The package in question is https://unpkg.com/resize-observer-polyfill

I believe the issue is in mkdocs-material-insiders/src/assets/javascripts/browser/element/size/_ /index.ts on line 79:
? watchScript("https://unpkg.com/resize-observer-polyfill")

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
- [Data Privacy Page](https://squidfunk.github.io/mkdocs-material/setup/ensuring-data-privacy/)

### Reproduction

[basic_test_project.zip](https://github.com/squidfunk/mkdocs-material/files/12446894/basic_test_project.zip)

### Steps to reproduce

1. activatre venv
2. run `python -m venv venv` and activate
3. install mkdocs-material-insiders
4. run `mkdocs serve --watch-theme  --strict`
5. check bundled JS `assets/javascripts/bundle.6fc684e3.min.js` `includes  https://unpkg.com/resize-observer-polyfill` even though privacy is enabled

### Browser

Chrome, Other

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting! Jup, that shouldn't happen. The problem stems from the fact that the file is extracted from another JavaScript file (the bundle) and the string is missing a file extension. Thus, we can't know to what the URL maps to, since JavaScript files might fetch other arbitrary files, including JSON, images, etc.

I replaced the URL with the resolved link (it's a 302 anyway) in [`de5be7c21`](https://github.com/squidfunk/mkdocs-material-insiders/commit/de5be7c214bcec6d57df3c507b7e0ccb2785a8be) which fixes the problem. If necessary, we might allow authors to provide expressions the privacy plugin should look for when resolving dependent assets in the future.

``` sh
$ mkdocs build
INFO    -  Cleaning site directory
INFO    -  Building documentation to directory: ...
INFO    -  Downloading external file: https://unpkg.com/resize-observer-polyfill/dist/ResizeObserver.js
INFO    -  Documentation built in 3.72 seconds
```

**Comment by squidfunk** [comment]
Released as part of [9.2.5+insiders-4.40.1](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.40.1).
